terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/main/java/org/terasoluna/gfw/common/validator/constraintvalidators/ByteSizeValidator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2013-2017 NTT DATA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package org.terasoluna.gfw.common.validator.constraintvalidators;
+
+import static org.terasoluna.gfw.common.validator.constraintvalidators.ConstraintValidatorsUtils.reportFailedToInitialize;
+
+import java.nio.charset.Charset;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.terasoluna.gfw.common.validator.constraints.ByteSize;
+
+/**
+ * Constraint validator class of {@link ByteSize} annotation.
+ * <p>
+ * Validate the {@link CharSequence}({@link String}, {@link StringBuilder}, etc ...) whose byte length must 
+ * be between the specified minimum and maximum. Determine the byte length By encoding the string in the specified charset.
+ * </p>
+ * @since 5.1.0
+ * @see ConstraintValidator
+ * @see ByteSize
+ */
+public class ByteSizeValidator implements
+                               ConstraintValidator<ByteSize, CharSequence> {
+
+    /**
+     * The charset used in parse to a string.
+     */
+    private Charset charset;
+
+    /**
+     * Byte length must be higher or equal to.
+     */
+    private long min;
+
+    /**
+     * Byte length must be lower or equal to.
+     */
+    private long max;
+
+    /**
+     * Initialize validator.
+     * @param constraintAnnotation annotation instance for a given constraint declaration
+     * @throws IllegalArgumentException failed to get a charset by name, or max is lower than min.
+     * @see javax.validation.ConstraintValidator#initialize(java.lang.annotation.Annotation)
+     */
+    @Override
+    public void initialize(ByteSize constraintAnnotation) {
+        try {
+            charset = Charset.forName(constraintAnnotation.charset());
+        } catch (IllegalArgumentException e) {
+            throw reportFailedToInitialize(e);
+        }
+        min = constraintAnnotation.min();
+        max = constraintAnnotation.max();

```

Comment:
- [ ] I prefer to add the check for `min >= 0` and `max >= 0` like implementation of `@Size` provided by Hibernate validator. (`@ByteMin` and `@ByteMax` have same issue)